### DISCUSSION TITLE
Send Notification on multiple device without loop?

### DISCUSSION THREAD

**Discussion by tarun-jangid** [relayrides/pushy#1012] [Q&A]
How can send notification on multiple device without loop?

**Comment by jchambers** [comment]
Pushy does not provide any "bulk send" affordances; you'll need to use a loop (or a `Stream` or some other kind of iteration strategy).

There are two reasons for this design decision:

1. The APNs protocol doesn't have any affordances for "bulk send" operations, so any "bulk send" Pushy offered would just be a loop anyhow and
2. Handling errors with bulk send operations is tricky, and I don't think there's a single correct way to do it that would satisfy all users

Hope that helps.
